helixbass/tsc-rust
# ast_type macro: generate basic node interface impl's for enum types

In this PR:
- generate `impl NodeInterface` and `impl ReadonlyTextRange` when `#[ast_type]` is applied to a non-leaf (enum) AST node type (eg `Expression`)

To test:
Things should still behave the same as of #9 
If you're curious to see the output of the macros you can `cargo install cargo-expand` and run eg `cargo expand --lib compiler::types > expanded` (from the repo root) and inspect the contents of the `expanded` file

Based on `ast-type-macro-node-interface`
